apps/jupyter/extensions/notebook-runner/src/task-importer.ts

```diff
@@ -0,0 +1,71 @@
+import JSZip from "jszip";
+import { CrtInternalFiles } from "./task-format";
+
+export interface CrtInternalTask {
+  taskTemplate: Blob;
+  studentTask: Blob;
+  autograder: Blob;
+  data: Map<string, Blob>;
+  gradingData: Map<string, Blob>;
+  src: Map<string, Blob>;
+  gradingSrc: Map<string, Blob>;
+}
+
+export interface ExternalCustomTask {
+  taskFile: Blob;
+  data: Map<string, Blob>;
+  gradingData: Map<string, Blob>;
+  src: Map<string, Blob>;
+  gradingSrc: Map<string, Blob>;
+}
+
+const extractFolder = async (
+  zip: JSZip,
+  prefix: string,
+): Promise<Map<string, Blob>> => {
+  const files = new Map<string, Blob>();
+
+  const folderPath = prefix.endsWith("/") ? prefix : `${prefix}/`;
+
+  for (const [path, file] of Object.entries(zip.files)) {
+    const relativePath = path.substring(folderPath.length);
+    if (relativePath) {
```

Comment:
When is `relativePath` falsey and what does that mean? Should we log, throw or ignore? Also we should probably document why it is fine to ignore if we decide to do that